### PR TITLE
SigV4 and AzureAD authentication support for `prometheus.remote_write`

### LINKED ISSUE #5346: prometheus.remote_write: Add support for Sigv4, AzureAD authentication mechanisms
### Request

The upstream Prometheus remote write configuration supports [sigv4](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#:~:text=SDK%2C%20use%20%60sigv4%3A%20%7B%7D%60.-,sigv4%3A,-%23%20The%20AWS%20region) and [azuread](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#:~:text=oauth2%2C%20or%20sigv4.-,azuread%3A,-%23%20The%20Azure%20Cloud) to be used as authentication mechanisms.

We should support the same in our prometheus.remote_write component.

### Use case

Being able to have feature-parity with the upstream remote_write configuration use cases.

### PR DESCRIPTION
**Pull request by ptodev**
#### PR Description

Adding SigV4 and AzureAD authentication support for `prometheus.remote_write`.

#### Which issue(s) this PR fixes

Fixes #5346

#### Notes to the Reviewer

Unlike other authentication methods, this one is not used in discovery processes since it's just specific to remote write.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

**Review by rfratto — APPROVED** [review]
Thanks! LGTM, pending docs review from Clayton.

**Review by clayton-cornell — APPROVED** [review]
Docs look ok.

**Pull request merged**
